agent/command/papertrail_trace_test.go

```diff
@@ -213,6 +213,184 @@ func TestPapertrailTrace(t *testing.T) {
+						filename: "test.tar.gz",
+						sha256:   "1b4f0e9851971998e732078544c96b36c3d01cedf7caa332359d6f1d83567014",
+					},
+				},
+			},
+			patterns: []string{"parent/test.tar.gz"},
+		},
+		"wildcard": {
+			files: []testFile{
+				{
+					path:    "test1.tar.gz",
+					content: "foo",
+					want: papertrailTraceFile{
+						filename: "test1.tar.gz",
+						sha256:   "2c26b46b68ffc68ff99b453c1d30413413422d706483bfa0f98a5e886266e7ae",
+					},
+				},
+				{
+					path:    "test2.tar.gz",
+					content: "bar",
+					want: papertrailTraceFile{
+						filename: "test2.tar.gz",
+						sha256:   "fcde2b2edba56bf408601fb721fe9b5c338d10ee429ea04fae5511b68fbf8fb9",
+					},
+				},
+			},
+			patterns: []string{"*.tar.gz"},
+		},
+		"wildcard-parent": {
+			files: []testFile{
+				{
+					path:    "parent/test1.tar.gz",
+					content: "foo",
+					want: papertrailTraceFile{
+						filename: "test1.tar.gz",
+						sha256:   "2c26b46b68ffc68ff99b453c1d30413413422d706483bfa0f98a5e886266e7ae",
+					},
+				},
+				{
+					path:    "parent/test2.tar.gz",
+					content: "bar",
+					want: papertrailTraceFile{
+						filename: "test2.tar.gz",
+						sha256:   "fcde2b2edba56bf408601fb721fe9b5c338d10ee429ea04fae5511b68fbf8fb9",
+					},
+				},
+			},
+			patterns: []string{"parent/*.tar.gz"},
+		},
+		"multiple-wildcard-parent": {
+			files: []testFile{
+				{
+					path:    "test-1/parent/test1.tar.gz",
+					content: "foo",
+					want: papertrailTraceFile{
+						filename: "test1.tar.gz",
+						sha256:   "2c26b46b68ffc68ff99b453c1d30413413422d706483bfa0f98a5e886266e7ae",
+					},
+				},
+				{
+					path:    "test-2/parent/test2.tar.gz",
+					content: "bar",
+					want: papertrailTraceFile{
+						filename: "test2.tar.gz",
+						sha256:   "fcde2b2edba56bf408601fb721fe9b5c338d10ee429ea04fae5511b68fbf8fb9",
+					},
+				},
+			},
+			patterns: []string{"*/parent/*.tar.gz"},
+		},
+	}
+
+	for name, tc := range cases {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			tempdir, err := os.MkdirTemp(os.TempDir(), "papertrail-get-files-test-*")
+			if err != nil {
+				t.Fatalf("make temp dir: %s", err)
+			}
+
+			t.Cleanup(func() { os.RemoveAll(tempdir) })
+
+			for _, f := range tc.files {
+				path := filepath.Join(tempdir, f.path)
+				parent := filepath.Dir(path)
+
+				if err := os.MkdirAll(parent, 0755); err != nil {
+					t.Fatalf("make parent dirs: %s", err)
+				}
+
+				if err := os.WriteFile(path, []byte(f.content), 0644); err != nil {
+					t.Fatalf("write file: %s", err)
+				}
+			}
+
+			got, err := getTraceFiles(tempdir, tc.patterns)
+			if err != nil {
+				t.Fatalf("get trace files: %s", err)
+			}
+
+			if len(got) != len(tc.files) {
+				t.Fatalf("got %d files, expected %d", len(got), len(tc.files))
+			}
+
+			for i, gotf := range got {
+				tcf := tc.files[i]
+
+				if gotf != tcf.want {
+					t.Fatalf("expected '%s', got '%s'", tcf.want, gotf)
+				}
+			}
+		})
+	}
+
+	t.Run("multiple-matches", func(t *testing.T) {
```

Comment:
Can we also test that matching disallowed file types (e.g. directories) produce an error?

Reply:
This won't produce an error, it'll just give us zero matches from `filepath.Glob`. Or are you thinking that zero matches *should* be an error? I can see the case for that, if so.

Reply:
Yeah that makes sense given that glob doesn't return directories, I just misunderstood the purpose of the path separator + "." check.